app/containers/scenes/MainMapScene.js

```diff
@@ -166,18 +166,20 @@ export default class MainMapScene extends Component {
 	}
 
 	_createTrailingMarkersForChallenge(challenge) {
-		var trailingMarkers = challenge.locations.slice(1).map((location, index) => (
-			{
-				id: challenge.id + "-" + (index+1),
-				title: location.title,
-				chalengeDescription: challenge.description,
-				latlng: {
-					latitude: location.latitude,
-					longitude: location.longitude
-				},
-			}
-		));
-		return trailingMarkers;
+		if(challenge.locations && challenge.locations.length > 1) {
```

Comment:
Most of this is unnecessary. Array.prototype.slice and Array.prototype.map work just fine on empty arrays and will return an empty array after execution so there is no need for an else clause to return "[]". In fact you can do `[].slice(10000).map(function(item) { return item * -2 })` and you will simply get back an empty array. There is no performance hit either as the map function is only called if there is actually an item to "map" to the new array. It's like a while loop that says "while(0 < 0)" so it is never executed and is essentially the same as the additional conditional check you added.

The challenge itself comes from our server and even if we were to allow a challenge to be created without any points, it would still return an empty array so challenge.locations should never be undefined. If for some strange reason locations were undefined however, it would cause an exception so we may want to add a default. Destructuring the required nested parameters and providing a default empty array is probably the cleanest way to do it. For example:

```
_createTrailingMarkersForChallenge({id, locations=[]}) {

    var trailingMarkers = locations.slice(1).map((location, index) => (
        {
            id: id + "-" + (index+1),
            title: location.title,
            latlng: {
                latitude: location.latitude,
                longitude: location.longitude
            },
        }
    ));
    return trailingMarkers;
}
```

But like I said, the other stuff can be safely removed.

Note: challengeDescription isn't actually being used for anything and is unnecessary for the marker so I removed it.